megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -1981,8 +1981,13 @@ private void unloadTransportedInfantry(MovePath path) {
             return;
         }
         
-        Entity movingEntity = path.getEntity();
-        Coords pathEndpoint = path.getFinalCoords();
+        final Entity movingEntity = path.getEntity();
+        final Coords pathEndpoint = path.getFinalCoords();
+        if (pathEndpoint == null) {
```

Comment:
This approach is incorrect. A null path endpoint means the entity isn't moving, so you can simply use the entity's current position instead. You still probably want an "is this point on the board" check though; technically off-board units don't get to participate in the movement phase, but with the movement phase code the way it is, you never know.

Reply:
It's also just a bandaid. I've reverted the null handling here and instead fixed the core issue, which would also be the cause of a bunch of other issues across the project (unofficial rule implementation issue)